doc/decisions/0012-repeating-answer-options.md

```diff
@@ -0,0 +1,274 @@
+
+### Dynamic answers based on a list source
+
+```json
+{
+  "question": {
+    "id": "percentage-of-research",
+    "title": "Some title",
+    "type": "Question",
+    "answers": [...],
+    "dynamic_answers": {
+      "values": {
+        "source": "list",
+        "identifier": "postcodes",
+        "selector": "items"
+      },
+      "answers": [
+        {
+          "description": {
+            "text": "Percentage of shopping at {transformed_value}",
+            "placeholders": [
+              {
+                "placeholder": "transformed_value",
+                "value": {
+                  "source": "answers",
+                  "identifier": "postcode"
+                }
+              }
+            ]
+          },
+          "id": "percentage-of-shopping",
+          "mandatory": false,
+          "type": "Percentage",
+          "maximum": {
+            "value": 100
+          },
+          "decimal_places": 2
+        }
+      ]
+    }
+  }
+}
+```
+- For dynamic answers driven by list collectors, for front end purposes each list item, at runtime the `list-item-id` or `selector` will be appended to the `id` used in the `dynamic_answers.answer` block, so that all answers have unique ids e.g. `"id": "percentage-of-shopping-{list_item_item_id}`. For storage we will just use the `list_item_id`.
+- Answer sources within a loop must be resolved using the `list_item_id` to resolve the current iterations value.
+
+### Dynamic answers based on a list of answers
+
+```json
+{
+  "question": {
+    "id": "percentage-of-research",
+    "title": "Some title",
+    "type": "Question",
+    "answers": [...],
+    "dynamic_answers": {
+      "values": {
+        "source": "answers",
+        "identifier": "checkbox-answer"
+      },
+      "answers": [
+        {
+          "description": {
+            "text": "Percentage of shopping at {transformed_value}",
+            "placeholders": [
+              {
+                "placeholder": "transformed_value",
+                "transforms": [
+                  {
+                    "transform": "option_label_from_value",
+                    "arguments": {
+                      "value": "self",
+                      "answer_id": "checkbox-answer"
+                    }
+                  }
+                ]
+              }
+            ]
+          },
+          "id": "percentage-of-shopping",
+          "mandatory": false,
+          "type": "Percentage",
+          "maximum": {
+            "value": 100
+          },
+          "decimal_places": 2
+        }
+      ]
+    }
+  }
+}
+```
+
+- `list_item_id` can now be an answer value or the output of a function, not just list collector ids.
```

Comment:
Mention the remove of spaces from values for it can be used as an id for storage.